bundles/com.zeligsoft.domain.dds4ccm/src/com/zeligsoft/domain/dds4ccm/utils/DDS4CCMMigrationModelTypeUtil.java

```diff
@@ -69,35 +92,268 @@ private static boolean migrateAll(Model model, boolean dryrun) {
+	 *          Preferred value of 'ConnectorType' to be used during migration 
+	 *            
+	 * @return pathmap of a connector library which corresponds to the input connectorType value  
+	 *         
+	 */
+	
+	private static String getPathmapConnectorLib(String preferredConnectorType){
+		
+		String pathmapConnectorLib = "";
+		
+		String pathmapAMIConnector = "pathmap://DDS4CCM_AMI_LIBRARIES/CCM_AMI.uml";
+		String pathmapCORBAConnector = "pathmap://DDS4CCM_CORBA_LIBRARIES/CCM_CORBA.uml";
+		
+		if(preferredConnectorType.equals(ConnectorType.AMI4CCM_Connector.name())){
+			pathmapConnectorLib = pathmapAMIConnector; 
+		}else if(preferredConnectorType.equals(ConnectorType.CORBA4CCM_Connector.name())){
+			pathmapConnectorLib = pathmapCORBAConnector;
+		}
+		
+		return pathmapConnectorLib;
+	}
+	
+	/**
+	 * Import a package represented by the input 'packageUri' in the input 'model' and return it. 
+	 * If the package is already imported, return the imported package.  
+	 * 
+	 * @param model
+	 *          model object where the package should be imported
+	 * @param packageUri
+	 * 			URI of the package object that is being imported          
+	 *            
+	 * @return the imported package object  
+	 *         
+	 */
+	private static Package getImportedConnectorPackage(Model model, URI packageUri){
+		
+		Package importedConnectorPackage = null;
+		boolean isAlreadyImported = false;
+	
+		List<PackageImport> listPackageImport = model.getPackageImports();
+		
+		for(PackageImport pi: listPackageImport){
+			
+			if(pi.getImportedPackage().eResource().getURI().equals(packageUri)){
+				isAlreadyImported = true;
+				importedConnectorPackage = pi.getImportedPackage();
+				break;
+			}
+		}
+		
+		if(!isAlreadyImported){
+			org.eclipse.emf.ecore.resource.Resource connectorPackageResource = model.eResource().getResourceSet().createResource(packageUri);			
+			
+			try {
+				connectorPackageResource.load(null);
+				importedConnectorPackage = (Package) connectorPackageResource.getContents().get(0);
+				importPackage(model, importedConnectorPackage);
+			} catch (IOException e1) {
+				// TODO Auto-generated catch block
+				e1.printStackTrace();
+			}			
+		}		
+		return importedConnectorPackage;		
+	}
+	
+	private static void importPackage(final Model model, final Package packageObj){
+		
+		org.eclipse.emf.ecore.resource.Resource modelResource = model.eResource();
+		final Package root = (Package) modelResource.getContents().get(0);
+				
+		TransactionalEditingDomain domain =	TransactionUtil.getEditingDomain(model);	
+		
+		Command importCommand = new RecordingCommand(domain, "import connector library"){
+			@Override
+			protected void doExecute(){
+				
+				root.createPackageImport(packageObj);
+			}
+		};
+		
+		try {
+			((TransactionalCommandStack)domain.getCommandStack()).execute(importCommand, null);
+		} catch (InterruptedException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
```

Comment:
If my comment above is correct, and no additional recording command and command execution is required, then this point is mute, but exceptions should never be suppressed like this. There are several possibilities for any exception:

- it is completely unexpected, in which case it should be logged to the Eclipse Error Log with a text stating that you believed this would never happen, but have been proven wrong. If the message ever appears in the error log, it means it is a bug.
- it is foreseeable that the exception could occur, but in this context, you do not have enough context to resolve it here. In that case, rethrow the exception (or never catch it in the first place). This might involve changing the method signature to include additional throws. If doing so it not permitted, you might have to wrap the exception in a RuntimeException subclass, or in an already permitted throwable exception.
- it is foreseeable that the exception could occur, and you are in a context where you can resolve the issue. In that case, you are justified in suppressing the exception, but I would expect to see the catch block doing something other than a printStackTrace().